flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java

```diff
@@ -1544,6 +1552,30 @@ private void notifyJobStatusChange(JobStatus newState) {
         }
     }
 
+    private void notifyJobStatusHooks(JobStatus newState, Throwable cause) {
+        JobID jobID = jobInformation.getJobId();
+        for (JobStatusHook hook : jobStatusHooks) {
+            try {
+                switch (newState) {
+                    case CREATED:
+                        hook.onCreated(jobID);
```

Comment:
maybe we should check cause is null for the branch other than FAILED.

Reply:
This is not necessary, only the FAILED branch needs to use cause